### ISSUE TITLE
aws-efs operator pod in CrashLoopBackOff state

### ISSUE THREAD

**Issue by pushkarmahakal1982** [openshift/aws-efs-operator#42]
Hello team, 

I am using OSD 4.9.0 on my production cluster. Post installing the aws-efs operator, I noticed that the efs-operator pod is in CrashLoopBackOff state. Below are the pods logs located under openshift-operators namespace. 
```
{"level":"info","ts":1636367532.7972922,"logger":"cmd","msg":"Operator Version: 0.0.1"}
{"level":"info","ts":1636367532.7973247,"logger":"cmd","msg":"Go Version: go1.16"}
{"level":"info","ts":1636367532.7973301,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1636367532.7973354,"logger":"cmd","msg":"Version of operator-sdk: v0.16.0"}
{"level":"info","ts":1636367532.7976806,"logger":"leader","msg":"Trying to become the leader."}
{"level":"info","ts":1636367535.6674385,"logger":"leader","msg":"Found existing lock with my name. I was likely restarted."}
{"level":"info","ts":1636367535.6674695,"logger":"leader","msg":"Continuing as the leader."}
{"level":"info","ts":1636367538.522777,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":"0.0.0.0:8383"}
{"level":"info","ts":1636367538.5229383,"logger":"cmd","msg":"Registering Components."}
{"level":"info","ts":1636367541.3824425,"logger":"cmd","msg":"Found. Checking whether update is needed.","resource":{"namespace":"openshift-operators","name":"efs-csi-sa"}}
{"level":"info","ts":1636367541.3824894,"logger":"cmd","msg":"No update needed."}
{"level":"info","ts":1636367541.3878217,"logger":"cmd","msg":"Found. Checking whether update is needed.","resource":{"name":"efs-csi-scc"}}
{"level":"info","ts":1636367541.3879697,"logger":"cmd","msg":"No update needed."}
{"level":"info","ts":1636367541.3955388,"logger":"cmd","msg":"Found. Checking whether update is needed.","resource":{"namespace":"openshift-operators","name":"efs-csi-node"}}
{"level":"info","ts":1636367541.395598,"logger":"cmd","msg":"Update needed. Updating..."}
{"level":"info","ts":1636367541.4047499,"logger":"cmd","msg":"Updated.","resource":{"namespace":"openshift-operators","name":"efs-csi-node"}}
{"level":"error","ts":1636367541.4048483,"logger":"cmd","msg":"Failed to retrieve.","resource":{"name":"efs.csi.aws.com"},"error":"no matches for kind \"CSIDriver\" in version \"storage.k8s.io/v1beta1\"","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nopenshift/aws-efs-operator/pkg/util.(*EnsurableImpl).Ensure\n\t/workdir/pkg/util/ensurable.go:87\nopenshift/aws-efs-operator/pkg/controller/statics.EnsureStatics\n\t/workdir/pkg/controller/statics/statics.go:201\nmain.main\n\t/workdir/cmd/manager/main.go:167\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"}
{"level":"info","ts":1636367541.4092035,"logger":"cmd","msg":"Found. Checking whether update is needed.","resource":{"name":"efs-sc"}}
{"level":"info","ts":1636367541.4092307,"logger":"cmd","msg":"No update needed."}
{"level":"error","ts":1636367541.4092348,"logger":"cmd","msg":"Couldn't bootstrap static resources","error":"Encountered 1 error(s) ensuring statics","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nmain.main\n\t/workdir/cmd/manager/main.go:168\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"}
```

Can someone please help me in this. 

Thanks, 
Pushkar

**Comment by 2uasimojo** [comment]
It looks like CSIDriver v1beta1 was removed in 4.9. Several older resource versions were, but either this one wasn't called out in the master list or we missed it when we were doing these fixes. (@camilamacedo86 FYI)
This will need someone to migrate the version of CSIDriver we use in aws-efs-operator. It doesn't have to be us -- community contributions are welcome.

**Comment by camilamacedo86** [comment]
Hi @2uasimojo and @pushkarmahakal1982, 

It seems that the operator `aws-efs operator` has as a dependency the operator `efs-operator`. Then, seems that it is trying to install or was installed on the cluster previously a version of the efs-operator which uses the removed APIs on 4.9. 

Then note that before you migrate your cluster to 4.9 you must ensure that all operators installed are compatible with 4.9 ( you will be notified in the biggest part of the cases from 4.8 to 4.9 in the case of not, however, it's recommended also check the events before upgrade the cluster to avoid this scenario)
 
To solve this problem is required to contact the authors and ask for them to publish a new version of their solutions that do not use the APIs which were removed from 4.9 and also ensure that their project depends on available distributions which are also workable on 4.9. More info: https://github.com/redhat-openshift-ecosystem/community-operators-prod/discussions/138

**Comment by jharrington22** [comment]
Hi all, this will be resolved once 0.0.8 is released https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/551

**Comment by jharrington22** [comment]
This specific issue was fixed in #44 but will require the above OperatorHub release to roll out to clusters.

**Comment by jharrington22** [comment]
This is now resolved on master and with the community operator version `0.0.8`.

Closing this issue.
